common/src/main/java/org/apache/sedona/common/Predicates.java

```diff
@@ -62,4 +64,82 @@ public static boolean dWithin(Geometry leftGeometry, Geometry rightGeometry, dou
             return leftGeometry.isWithinDistance(rightGeometry, distance);
         }
     }
+
+    /**
+     * Checks if a geometry crosses the International Date Line.
+     *
+     * @param geometry The geometry to check.
+     * @return True if the geometry crosses the Date Line, false otherwise.
+     */
+    public static boolean crossesDateLine(Geometry geometry) {
+        if (geometry == null || geometry.isEmpty()) {
+            return false;
+        }
+
+        AtomicBoolean crossesDateLine = new AtomicBoolean(false);
+
+        CoordinateSequenceFilter filter = new CoordinateSequenceFilter() {
+            private Coordinate previous = null;
+
+            @Override
+            public void filter(CoordinateSequence seq, int i) {
+                if (i == 0) {
+                    previous = seq.getCoordinateCopy(i);
+                    return;
+                }
+
+                Coordinate current = seq.getCoordinateCopy(i);
+                if (Math.abs(current.x - previous.x) > 180) {
+                    crossesDateLine.set(true);
+                }
+
+                previous = current;
+            }
+
+            @Override
+            public boolean isDone() {
+                return crossesDateLine.get();
+            }
+
+            @Override
+            public boolean isGeometryChanged() {
+                return false;
+            }
+        };
+
+        if (geometry instanceof GeometryCollection) {
```

Comment:
Why explicitly do this for `GeometryCollection`? Note that: `GeometryCollection` can have nested `GeometryCollection` inside. We should avoid implementing our own logic to iterate GeometryCollection. If we really want to do it, this should be done via a recursive function.

Reply:
I initially wanted to make sure Every geometry within the `GeometryCollection` is handled separately because `crossesDateLine` should return true if any geometry is crossing the dateline.
You're right, explicit handling for `GeometryCollection` was not required, `CoordinateSequenceFilter` is able to handle this well. Have pushed the changes.